src/pages/keyword-generator/KeywordGenerator.vue

```diff
@@ -1,9 +1,99 @@
 <script setup>
+import { ref } from 'vue';
+const nGrams = ref([]);
+const text = ref('');
+
+const generateNGram = (keywords, n) => {
+  const result = new Set();
+  const words = keywords.split(" ");
+  for (let i = 0; i <= words.length - n; i++) {
+    const nGram = words.slice(i, i + n).join(' ');
+    result.add(nGram);
+  }
+  return Array.from(result);
+}
+
+const generateNGrams = () => {
+  nGrams.value = [];
+  if (!text.value.trim()) { return; }
+  for (let i = 1; i <= 3; i++) {
+    const nGramList = generateNGram(text.value, i);
+    if (nGramList.length) {
+      nGrams.value.push({ i, keywords: nGramList });
+    }
+  }
+}
 </script>
+
 <template>
-    <div class="ma-keywords-generator">
-        <div class="ma-header">
-            <span>Keyword Generator</span>
-        </div>
+  <div class="ma-keywords-generator">
+    <div class="ma-header">
+      <h3>Keyword Generator</h3>
     </div>
-</template>
\ No newline at end of file
+
+    <div class="ma-body">
+      <textarea v-model="text" placeholder="Enter a text here..." class="text-input"></textarea>
+      <button class="generate-button" @click="generateNGrams">Generate</button>
+    </div>
+
+    <div class="display">
+      <h3> Generated Keywords </h3>
+      <div v-for="nGram in nGrams" :key="nGram.i">
+        <h4>{{ nGram.i }}-gram keywords:</h4>
+        <ul>
+          <li v-for="keyword in nGram.keywords" :key="nGram.i + index">{{keyword}}</li>
+        </ul>
+      </div>
+    </div>
+
+  </div>
+</template>
+<style scoped>
```

Comment:
To keep the readability of the code at a high level, we must pay attention to the code layout. We should avoid extra spaces. We should always leave space to the left and right of the operators.